### PR TITLE
remove most of THREE dependencies in vendor (fixes #720)

### LINKED ISSUE #720: Remove the `vendor` dir and use the modules in three.js and in NPM.

### PR DESCRIPTION
**Pull request by cvan**

### CODE REVIEW

File: lib/three.js [hunk 1]
```diff
@@ -6,10 +6,23 @@ if (THREE.TextureLoader) {
...
+require('../node_modules/three-dev/examples/js/loaders/OBJLoader');
+
+// THREE.ColladaLoader
+require('../node_modules/three-dev/examples/js/loaders/ColladaLoader');
+
+// THREE.ColladaLoader
+require('../node_modules/three-dev/examples/js/loaders/ColladaLoader');
+
+// THREE.Raycaster
+require('../lib/vendor/Raycaster');
```
Comment (cvan) [inline lib/three.js]: @ngokevin if you can change your `three-dev`'s `package.json` the `directories` to include `src/core`, we could reference `Raycaster.js` so we wouldn't need it here.

Comment (cvan) [inline lib/three.js]: it's weird that `Raycaster` isn't included since a lot of the examples rely on it.

Comment (ngokevin) [inline lib/three.js]: ok will do

**Comment by ngokevin** [comment]
something's up with the tests

**Comment by cvan** [comment]
> something's up with the tests

if I revert my local "0.74.0-beta5" to "0.74.0-beta3" then the tests fail. so I guess that's why; the files moved.

**Comment by cvan** [comment]
specifically, `examples` doesn't exist in `beta3`

**Comment by dmarcos** [comment]
what do we gain with this? each time I want to use a three js module I have to do PR in three-dev? I see this like becoming a three js fork and becoming a burden we have to also maintain

**Comment by dmarcos** [comment]
colladaloader require is duplicated

**Comment by ngokevin** [comment]
we're already maintaining several files in the vendor . i think this is easier since we just point to where the file is in three.js.

`three-dev` doesn't make modifications to three, we just use it to publish the version we want to be on

**Comment by cvan** [comment]
> what do we gain with this? each time I want to use a three js module I have to do PR in three-dev? I see this like becoming a three js fork and becoming a burden we have to also maintain

each time you use a new three.js module, you reference it the same way I've done in `THREE.js`.

we're making manual modifications to these `vendor/` files (`module.exports = function (THREE) { … }`, for example), which are unnecessary (since we're already polluting the `window.THREE` global now, since a few months back). and every time we update to the latest version of `THREE`, we have to _remember_ and _manually_ edit these `vendor/` files - and doing copying + pasting + diffing by hand seems riskier if we're already using `three` as a module.

**Comment by dmarcos** [comment]
What's going on with the tests?

**Comment by cvan** [comment]
> What's going on with the tests?

what I said above: https://github.com/aframevr/aframe-core/pull/721#issuecomment-165935693

this is dependent on an upgrade to three.js's dev (#722, formerly #719)

**Comment by dmarcos** [comment]
I merged https://github.com/aframevr/aframe-core/pull/722. Can we rebase to see if it fixes the tests?

**Comment by cvan** [comment]
will do, thanks

**Pull request merged**